pkg/tfbridge/property_path.go

```diff
@@ -0,0 +1,127 @@
+package tfbridge
+
+import (
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge/info"
+	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+)
+
+type propertyPath resource.PropertyPath
+
+func isForceNew(tfs shim.Schema, ps *SchemaInfo) bool {
+	return (tfs != nil && tfs.ForceNew()) ||
+		(ps != nil && ps.ForceNew != nil && *ps.ForceNew)
+}
+
+func newPropertyPath(root resource.PropertyKey) propertyPath {
+	return propertyPath{string(root)}
+}
+
+func (k propertyPath) String() string {
+	return resource.PropertyPath(k).String()
+}
+
+func (k propertyPath) Key() detailedDiffKey {
+	return detailedDiffKey(k.String())
+}
+
+func (k propertyPath) append(subkey interface{}) propertyPath {
+	return append(k, subkey)
+}
+
+func (k propertyPath) Subpath(subkey string) propertyPath {
+	return k.append(subkey)
+}
+
+func (k propertyPath) Index(i int) propertyPath {
+	return k.append(i)
+}
+
+func (k propertyPath) IsReservedKey() bool {
+	leaf := k[len(k)-1]
+	return leaf == "__meta" || leaf == "__defaults"
+}
+
+func lookupSchemas(
+	path propertyPath, tfs shim.SchemaMap, ps map[string]*info.Schema,
+) (shim.Schema, *info.Schema, error) {
+	schemaPath := PropertyPathToSchemaPath(resource.PropertyPath(path), tfs, ps)
+	return LookupSchemas(schemaPath, tfs, ps)
+}
+
+// walkPropertyValue walks a property value and calls the visitor function for each path in the property value.
+func walkPropertyValue(
```

Comment:
We have some walkers for PropertyValue elsewhere.

https://github.com/pulumi/pulumi-terraform-bridge/blob/master/unstable/propertyvalue/propertyvalue.go#L21

Transform is also a walker if f does not edit the value. 

I know Go is a little bit more of a "let's do a little more code duplication than too much restrictive sharing" language so perhaps you looked there and it didn't fit the bill, but highlighting this.

Ideally the walkers would live in core alongside of PropertyValue definition itself.

Reply:
I didn't find anything like that so I wrote my own - happy to re-use any existing code here.

Reply:
PTAL at the linked code and consider 🙏

Reply:
addressed in the previous PR in the stack: https://github.com/pulumi/pulumi-terraform-bridge/pull/2496